---
title: "TD2 - Modèle mixte"
author: "Emilie Mathian"
date: "01/10/2018"
output:
  html_document:
    toc: true
    number_sections: true
    mathjax: local
    self_contained: false
---
<style>
.alert-info {
  color: rgb(49,112,143) !important;
}
</style>


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Librairie
```{r eval=TRUE, echo=FALSE}
library(DESeq2)
library(data.table)
library(dplyr)
library(ade4)
```


## Importation des données
**Ce Rmarkdown a été crée à partir du dossier MESO_DATA**
```{r eval=TRUE, echo=TRUE}
metadata <- read.table("metaBuenoTCGAred_FGS.txt", sep = "\t", dec="." , header = TRUE,   quote="")
```


`metadata` contient 284 échantillons avec les données principales PC1, PC2, Type , Sample ...

```{r eval=TRUE, echo=TRUE}
attribute <- load("Rdata8nov2018.RData")
```
**Remarque : dinomesomics RNA reads count pour 284 ech**

```{r eval=TRUE, echo=TRUE}
suptable1<- read.table("SupTables_R1_20190418/TableS1_Characteristics_of_gene-Tableau 1.csv", sep = ";", dec="," , header = TRUE,   quote="")
head(suptable1,4)
```



# Ceration du tableau des coordonnées 


```{r eval=TRUE, echo=TRUE}
Coordinates <- data.frame( sample = suptable1$Samples , x = suptable1$Dimension.1 , y = suptable1$Dimension.2)

Coordinates2 <- Coordinates 
Coordinates2$y <- Coordinates2$y * -1
```

## Correction du nom des échantillons dans Groups Bueno TCGA red :
```{r eval=TRUE, echo=TRUE}
Diff  = setdiff(groupsBuenoTCGAred$Sample , suptable1$Samples)
```
**Attention qq nom d'échantillons ne sont pas identiques entre suptable1 et groupsBuenoTCGAred**


```{r eval=TRUE, echo=TRUE}

groupsBuenoTCGAred_2= groupsBuenoTCGAred
for (i in  1:284){
  if (groupsBuenoTCGAred$Sample[i] %in% Diff){
    groupsBuenoTCGAred_2$Sample[i] = paste("M", groupsBuenoTCGAred$Sample[i], sep="" )
  }
}
setdiff(groupsBuenoTCGAred_2$Sample, suptable1$Samples)
```

# Definition des attributs


```{r eval=TRUE, echo=TRUE}

TILS <- data.frame( sample = suptable1$Samples , B = suptable1$B , Macrophages.M1 = suptable1$Macrophages.M1 , Macrophages.M2 = suptable1$Macrophages.M2 , Monocytes = suptable1$Monocytes, Neutrophils = suptable1$Neutrophils , NK.cells = suptable1$Macrophages.M2 ,T.cells.CD4 = suptable1$T.cells.CD4  , T.cells.CD8 = suptable1$T.cells.CD8 , Tregs = suptable1$Tregs , Dendritic.cells = suptable1$Dendritic.cells )


ClinicalAttributes <- data.frame( sample = groupsBuenoTCGAred_2$Sample, Asbestos =  groupsBuenoTCGAred_2$Asbestos , Sex= groupsBuenoTCGAred_2$Sex , Type = groupsBuenoTCGAred_2$Type , Age = groupsBuenoTCGAred_2$Age , Survival = groupsBuenoTCGAred_2$Survival ) 

```
Ajout des données subtype.fgs et type.fgs

```{r eval=TRUE, echo=FALSE}
Subtype_Type_revised =  data.frame( sample = as.character(metadata$Sample) , Subtype_FGS = metadata$Subtype.FGS , Type_FGS = metadata$Type.FGS , Necrosis = metadata$Necrosis, grade = metadata$Grade, stringsAsFactors=FALSE)
str(Subtype_Type_revised)
Diff= setdiff(Subtype_Type_revised$sample, suptable1$Samples); Diff
for (i in  1:284){
  if (Subtype_Type_revised$sample[i] %in% Diff){
       Subtype_Type_revised$sample[i] = paste("M",as.character( Subtype_Type_revised$sample[i]), sep="" )
  }
}


```

# Genes d'intérêt


* VISTA  : ENSG00000107738.19
* CD8A   : ENSG00000153563.15
* VEGFR1 : ENSG00000102755.10
* VEGFR2 : ENSG00000128052.8
* VEGFR3 : ENSG00000113721.13
* VEGFC  : ENSG00000150630.3
* PDGFRB : ENSG00000113721.13
* PD1    : ENSG00000188389.10
* PDL1  : ENSG00000120217.13
* CTLA4  : ENSG00000163599.14
* TIM3  : ENSG00000135077.8
* LAG3  : ENSG00000089692.8

Création de vecteurs de comptage pour ces gènes  :

```{r eval=TRUE, echo=TRUE}
VISTA = c(dinomesomics["ENSG00000107738.19",])
CD8A = c(dinomesomics["ENSG00000153563.15",])
VEGFR1 = c(dinomesomics["ENSG00000102755.10",])
VEGFR2 = c(dinomesomics["ENSG00000128052.8",])
VEGFR3 = c(dinomesomics["ENSG00000037280.15",])
VEGFC = c(dinomesomics["ENSG00000150630.3",])
PDGFRB = c(dinomesomics["ENSG00000113721.13",])
PD1  = c(dinomesomics["ENSG00000188389.10",])
PDL1 = c(dinomesomics["ENSG00000120217.13",])
CTLA4 = c(dinomesomics["ENSG00000163599.14",])
TIM3 = c(dinomesomics["ENSG00000135077.8",])
LAG3 = c(dinomesomics["ENSG00000089692.8",])
```

Création d'une clée :
```{r eval=TRUE, echo=TRUE}
gene_interet <- data.frame(VISTA,CD8A,VEGFR1, VEGFR2,VEGFR3, VEGFC , PDGFRB,PD1,PDL1, CTLA4 , TIM3 , LAG3 )

ID = sampleFilesred
New_ID = c()
for (i in 1:length(ID)){
  New_ID[i] =  sub("_count.txt","", ID[i])  
}



metadataID = data.frame(ID = New_ID, sample = suptable1$Sample)
gene_interet = setDT(gene_interet, keep.rownames = TRUE)[]
colnames(metadataID)[1] <- "ID"
colnames(gene_interet)[1] <- "ID"

gene_interet_samples_names <- merge(metadataID, gene_interet, by="ID")
gene_interet2 <- gene_interet_samples_names[,-1]
```


# Construction du data frame des atributs



```{r eval=TRUE, echo=TRUE}

Attributes1 <- merge(TILS, ClinicalAttributes, by="sample")

Attributes2 <- merge(Attributes1, gene_interet2, by="sample")

Attributes3 <- merge(Attributes2, Subtype_Type_revised , by = "sample")
str(Attributes3)
# Convert factor to char
Attributes3.2 <- Attributes3
Attributes3.2$sample = as.character(Attributes3$sample)
Attributes3.2$Asbestos =  as.character(Attributes3$Asbestos)
Attributes3.2$Sex =  as.character(Attributes3$Sex)
Attributes3.2$Type =  as.character(Attributes3$Type)
Attributes3.2$Subtype_FGS=  as.character(Attributes3$Subtype_FGS)
Attributes3.2$Type_FGS=  as.character(Attributes3$Type_FGS)
#str(Attributes3.2)
Attributes3$Type = as.character(Attributes3$Type)
```


```{r eval=TRUE, echo=FALSE}

Attributes3$Type[is.na(Attributes3$Type)] <- "Diffuse_NOS"
```

# Remplacement des valeurs manquantes pas 0

```{r eval=TRUE, echo=TRUE}
Attributes3.2[is.na(Attributes3.2)] <- 0
```

# Survie en mois

```{r eval=TRUE, echo=TRUE}

Attributes3$Survival<- Attributes3$Survival*12
```


# Ecriture des tableau 
```{r eval=TRUE, echo=TRUE}
write.table(Coordinates, file='Coordinates_PCA_f1.tsv', quote=FALSE, sep='\t', row.names = F)
write.table(Coordinates2, file='Coordinates_PCA_f2_invY.tsv', quote=FALSE, sep='\t', row.names = F)
write.table(Attributes3, file='Attributes_PCA_f1.tsv', quote=FALSE, sep='\t', row.names = F)

```


# Test sur le type de matrices
```{r eval=TRUE, echo=TRUE}
#Sample_coord <- read.table("Coordinates_PCA_f1.tsv", header = T)

#Attributes3.sample<-data.frame(matrix(ncol = 32))
#colnames(Attributes3.sample) <- colnames(Attributes3)
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M23PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M35PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M54PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M42PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M36PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M70PT") )
#Attributes3.sample <- rbind(Attributes3.sample, subset(Attributes3, sample == "M18PT") )

#Attributes3.sample <- Attributes3.sample[-1,]
#Attributes3.sample<-Attributes3.sample[,1:5]
#write.table(Attributes3.sample, file='Attributes_PCA_sample.tsv', quote=FALSE, sep='\t', row.names = F)

```

# Donnée rna_Seq
```{r eval=TRUE, echo=FALSE}

#dinomesomics_t = as.data.frame(t(dinomesomics) )
#dinomesomics_t = setDT(dinomesomics_t , keep.rownames = TRUE)[]
#colnames(dinomesomics_t)[1] <- "ID"
#dinomesomics_t_sample = merge(metadataID,dinomesomics_t, by ='ID')
#dinomesomics_t_sample = dinomesomics_t_sample[,-1]
#dinomesomics_t_sample = t(dinomesomics_t_sample)
#write.table(dinomesomics_t_sample, file='feature_data.tsv', quote=FALSE, sep='\t', row.names = F, col.names = F)
```



# Cohorte des 187 échantillons avec les données d'Immunohistochimie

## Importation des données

**Attention GNT57 a été enlevé manuelement du fichier `IHC_Lecture_LM_20181116_333_2.csv`**
```{r eval=TRUE, echo=TRUE}
IHC_333.2 <- read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_333_2.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_334.2 <- read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_334_2.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_351.1 <- read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_351_1.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_351.1 <- read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_351_1.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_352.1 <-  read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_352_1.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_353.1 <-  read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_353_1.csv", sep = ";", dec="." , header = TRUE, na.strings=c("","NA"))
IHC_354.1 <-   read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_354_1.csv", sep = ";", dec="." , header = TRUE,na.strings=c("","NA"))

#IHC_411.1 <-  read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_411_1.csv", sep = ";", dec="." , header = TRUE,na.strings=c("","NA"))
#IHC_412.1 <- read.csv("../IHC_Lecture_LM_20181121/IHC_Lecture_LM_20181116_412_1.csv", sep = ";", dec="." , header = TRUE,na.strings=c("","NA"))
```


## Changement du nom  des échantillons


```{r eval=TRUE, echo=TRUE}
change_sample_name_IHC_file <- function(data){
  TGCA_ID <- c()
  for (i in 1:dim(data)[1]){
    if (is.na(data$IARC_ID[i])==F){
      if (sum(is.na(data[i,4:8])) != 5){
        withM <- test_ID <- sub("GNT", "M", data$IARC_ID[i])[1]    
        g_expr = gregexpr(pattern ='_',withM)[[1]][1]
        if  (g_expr !=  -1){
          TGCA_ID[i] <- paste(substr(withM, 1, g_expr-1), "PT", sep="")[1]
        }
        else{
         TGCA_ID[i] <- paste(withM, "PT", sep="")[1]
       }
      }
      else{
        TGCA_ID[i] <- NA
      }
    }
    else{
      TGCA_ID[i] <- NA
    }
  }
  return(TGCA_ID)
}

IHC_333.2["Sample"] =  change_sample_name_IHC_file(IHC_333.2)
IHC_334.2["Sample"]  = change_sample_name_IHC_file(IHC_334.2)
IHC_351.1["Sample"]  = change_sample_name_IHC_file(IHC_351.1)
colnames(IHC_352.1)[2] <- "IARC_ID"
IHC_352.1["Sample"]  = change_sample_name_IHC_file(IHC_352.1)
colnames(IHC_353.1)[2] <- "IARC_ID"
IHC_353.1["Sample"]  = change_sample_name_IHC_file(IHC_353.1)
colnames(IHC_354.1)[2] <- "IARC_ID"
IHC_354.1["Sample"]  = change_sample_name_IHC_file(IHC_354.1)
#colnames(IHC_411.1)[2] <- "IARC_ID"
#IHC_411.1["Sample"]  = change_sample_name_IHC_file(IHC_411.1)
#colnames(IHC_412.1)[2] <- "IARC_ID"
#IHC_412.1["Sample"]  = change_sample_name_IHC_file(IHC_412.1)
```


```{r eval=TRUE, echo=TRUE}
#U_333.2 <-length(unique(IHC_333.2$Sample))
#U_334.2 <-length(unique(IHC_334.2$Sample))
#U_351.1 <-length(unique(IHC_351.1$Sample))
#U_352.1 <-length(unique(IHC_352.1$Sample))
#U_353.1 <-length(unique(IHC_353.1$Sample))
#U_354.1 <-length(unique(IHC_354.1$Sample))
#U_411.1 <-length(unique(IHC_411.1$Sample))
#U_412.1 <-length(unique(IHC_412.1$Sample))
tot_sample <- c(IHC_333.2$Sample , IHC_334.2$Sample, IHC_351.1$Sample, IHC_352.1$Sample,  IHC_353.1$Sample , IHC_354.1$Sample     ) #IHC_411.1$Sample , IHC_412.1$Sample 

#length(unique(tot_sample))
uniq_tot_sample= unique(tot_sample)[-1]
```


**Attention : 108  échantillons moins le NA**

## Pour CD8 et VEGFR3M passage en pourcentage 

```{r eval=TRUE, echo=TRUE}
CD8_VEGFR3_Pourcent <- function(data){
  CD8_p <- c()
  VEGFR3M_p <- c()
  for (i in 1:dim(data)[1]){
    if (is.na(data$CD8[i]) == F){
      if (data$CD8[i]==0){
        CD8_p[i]<- 0
      }
      else if(data$CD8[i]==1){
        CD8_p[i]<- 25
      }
      else if(data$CD8[i]==2){
        CD8_p[i]<- 50
      }
      else if(data$CD8[i]==3){
        CD8_p[i]<- 75
      }
      else if(data$CD8[i]==4){
        CD8_p[i]<- 100
      }
      else {
        CD8_p[i]<- NA
      }
    }
    else{
       CD8_p[i]<- NA
    }
    if (is.na(data$VEGFR3M[i])==F){
    if (data$VEGFR3M[i]==0){
      VEGFR3M_p[i]<- 0
    }
    else if(data$VEGFR3M[i]==1){
      VEGFR3M_p[i]<- 25
    }
    else if(data$VEGFR3M[i]==2){
      VEGFR3M_p[i]<- 50
    }
    else if(data$VEGFR3M[i]==3){
      VEGFR3M_p[i]<- 75
    }
    else if(data$VEGFR3M[i]==4){
      VEGFR3M_p[i]<- 100
    }
    else {
      VEGFR3M_p[i]<- NA
    }
    }
    else{
       VEGFR3M_p[i]<- NA
    }
  }
  CD8_VEGFR3_Pourcent_df <- data.frame(CD8_p, VEGFR3M_p) 
  return (CD8_VEGFR3_Pourcent_df)
}


IHC_333.2["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_333.2)[,1]
IHC_333.2["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_333.2)[,2]

IHC_334.2["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_334.2)[,1]
IHC_334.2["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_334.2)[,2]   

IHC_351.1["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_351.1)[,1]
IHC_351.1["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_351.1)[,2]    

IHC_352.1["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_352.1)[,1]
IHC_352.1["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_352.1)[,2] 


IHC_353.1["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_353.1)[,1]
IHC_353.1["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_353.1)[,2] 

IHC_354.1["CD8_p"]<-CD8_VEGFR3_Pourcent(IHC_354.1)[,1]
IHC_354.1["VEGFR3M_p"]<-CD8_VEGFR3_Pourcent(IHC_354.1)[,2] 

                                      
```


# Création d'une table finale 
```{r eval=TRUE, echo=TRUE}
IHC_all <- data.frame()

IHC_333.2_2 <- data.frame( Sample = IHC_333.2$Sample , VISTA = IHC_333.2$VISTA, PDL1 = IHC_333.2$PDL1 , VEGFR3 = IHC_333.2$VEGFR3M_p , CD8 = IHC_333.2$CD8_p , VEGFR2 = IHC_333.2$VEGFR2)

IHC_all <- rbind(IHC_333.2_2 ,IHC_all)


IHC_334.2_2<- data.frame( Sample = IHC_334.2$Sample , VISTA = IHC_334.2$VISTA, PDL1 = IHC_334.2$PDL1 , VEGFR3 = IHC_334.2$VEGFR3M_p , CD8 = IHC_334.2$CD8_p , VEGFR2 = IHC_334.2$VEGFR2)

IHC_all <- rbind(IHC_334.2_2 ,IHC_all)

IHC_351.1_2<- data.frame( Sample = IHC_351.1$Sample , VISTA = IHC_351.1$VISTA, PDL1 = IHC_351.1$PDL1 , VEGFR3 = IHC_351.1$VEGFR3M_p , CD8 = IHC_351.1$CD8_p , VEGFR2 = IHC_351.1$VEGFR2)

IHC_all <- rbind(IHC_351.1_2 ,IHC_all)

IHC_352.1_2<- data.frame( Sample = IHC_352.1$Sample , VISTA = IHC_352.1$VISTA, PDL1 = IHC_352.1$PDL1 , VEGFR3 = IHC_352.1$VEGFR3M_p , CD8 = IHC_352.1$CD8_p , VEGFR2 = IHC_352.1$VEGFR2)
IHC_all <- rbind(IHC_352.1_2 ,IHC_all)

IHC_353.1_2<- data.frame( Sample = IHC_353.1$Sample , VISTA = IHC_353.1$VISTA, PDL1 = IHC_353.1$PDL1 , VEGFR3 = IHC_353.1$VEGFR3M_p , CD8 = IHC_353.1$CD8_p , VEGFR2 = IHC_353.1$VEGFR2)
IHC_all <- rbind(IHC_353.1_2 ,IHC_all)

IHC_354.1_2<- data.frame( Sample = IHC_354.1$Sample , VISTA = IHC_354.1$VISTA, PDL1 = IHC_354.1$PDL1 , VEGFR3 = IHC_354.1$VEGFR3M_p , CD8 = IHC_354.1$CD8_p , VEGFR2 = IHC_354.1$VEGFR2)
IHC_all <- rbind(IHC_354.1_2 ,IHC_all)

#IHC_411.1_2<- data.frame( Sample = IHC_411.1$Sample , VISTA = IHC_411.1$VISTA, PDL1 = IHC_411.1$PDL1 , VEGFR3 = IHC_411.1$VEGFR3M_p , CD8 = IHC_411.1$CD8_p , VEGFR2 = IHC_411.1$VEGFR2)
#IHC_all <- rbind(IHC_411.1_2 ,IHC_all)


#IHC_412.1_2<- data.frame( Sample = IHC_412.1$Sample , VISTA = IHC_412.1$VISTA, PDL1 = IHC_412.1$PDL1 , VEGFR3 = IHC_412.1$VEGFR3M_p , CD8 = IHC_412.1$CD8_p , VEGFR2 = IHC_412.1$VEGFR2)
#IHC_all <- rbind(IHC_412.1_2 ,IHC_all)

```

## Moyenne par échantillon

```{r eval=TRUE, echo=TRUE}
IHC_moy_df <- data.frame(sample =uniq_tot_sample  ,  VEGFR2.IHC = rep(0,length(uniq_tot_sample)), VEGFR3M.IHC = rep(0,length(uniq_tot_sample)), PDL1.IHC = rep(0,length(uniq_tot_sample)), CD8.IHC = rep(0,length(uniq_tot_sample)), VISTA.IHC = rep(0,length(uniq_tot_sample)))

IHC_moy_sample <- function(data, res_df){
  for (i in 1:length(uniq_tot_sample)){
    c_sample_id = IHC_moy_df$sample[i]
    c_CD8_m =  mean(as.numeric(IHC_all$CD8[which(IHC_all$Sample== c_sample_id)]), na.rm=TRUE)
    res_df[i,5] <-c_CD8_m # Ok
    c_VISTA =  mean(as.numeric(IHC_all$VISTA[which(IHC_all$Sample== c_sample_id)]), na.rm=TRUE)
    res_df[i,6] <-c_VISTA 
    c_PDL1 =  mean(as.numeric(IHC_all$PDL1[which(IHC_all$Sample== c_sample_id)]), na.rm=TRUE)
    res_df[i,4] <-c_PDL1 #OK
    c_VEGFR3 =  mean(as.numeric(IHC_all$VEGFR3[which(IHC_all$Sample== c_sample_id)]), na.rm=TRUE)
    res_df[i,3] <-c_VEGFR3
    c_VEGFR2 =  mean(as.numeric(IHC_all$VEGFR2[which(IHC_all$Sample== c_sample_id)]), na.rm=TRUE)
    res_df[i,2] <- c_VEGFR2 
  }
  return(res_df)
    
}
Test = IHC_moy_sample(IHC_all , IHC_moy_df)
IHC_moy_df = IHC_moy_sample(IHC_all , IHC_moy_df)
# Supression des données pour lesquels les données génomiques sont manquantes
v1 = which(IHC_moy_df$sample=="M654PT")
IHC_moy_df = IHC_moy_df [-v1,]
v2 = which(IHC_moy_df$sample=="M78PT")
IHC_moy_df = IHC_moy_df [-v2,]
v3 = which(IHC_moy_df$sample=="M601PT")
IHC_moy_df = IHC_moy_df [-v3,]
```


## Observation des corrélation
```{r eval=TRUE, echo=TRUE}
IHC_attribute = merge(IHC_moy_df, Attributes3, by="sample", all=FALSE)

cor(IHC_attribute$CD8.IHC, IHC_attribute$CD8A, use = "complete.obs")
cor(IHC_attribute$VEGFR2.IHC, IHC_attribute$VEGFR2, use = "complete.obs")
cor(IHC_attribute$PDL1.IHC, IHC_attribute$PDL1, use = "complete.obs")
cor(IHC_attribute$VEGFR3M.IHC, IHC_attribute$VEGFR3, use = "complete.obs")
cor(IHC_attribute$VISTA.IHC, IHC_attribute$VISTA, use = "complete.obs")
```

# Write Attribut avec IHC
```{r eval=TRUE, echo=TRUE}
Attributes4 <- merge(Attributes3 , IHC_moy_df , by= "sample" , all= TRUE) 
colnames(Attributes4)[2] <- "B.Cells"
colnames(Attributes4)[29] <- "Subtype"
colnames(Attributes4)[30] <- "Type_revised"
write.table(Attributes4, file='Attributes_4_with_IHC.tsv', quote=FALSE, sep='\t', row.names = F)
```


# Feature_DATA with the largest variance

```{r eval=TRUE, echo=TRUE}
###### WRONG  !!!!

#data_lv  = merge(dinomesomics , dataNOS , by=0) # Merge by row names
#data_lv = t(data_lv)
#colnames(data_lv) = data_lv[1,]
#data_lv = data_lv[-1,]
#data_lv_2 = as.data.frame(data_lv, colnames=T, rownames= T) ;
#data_lv_2["ID"] = rownames(data_lv)
#data_lv_2 = merge(metadataID, data_lv_2 , by="ID")

 
#write.table(data_lv_2, file='feature_data_with_lv_2.tsv', quote=FALSE, sep='\t', row.names = F, col.names = F)
```


```{r eval=TRUE, echo=TRUE}
data_lv  = merge(data , dataNOS , by=0) # Merge by row names
dim(data_lv)
f_colnames = data_lv$Row.names
data_lv = as.data.frame(t(data_lv), col.names =FALSE )
colnames(data_lv) <- f_colnames 
data_lv  <- data_lv[-1,]
#data_lv = setDT(data_lv , keep.rownames = TRUE)[]
#data_lv = data_lv[-1,]
#colnames(data_lv)[1] <- "ID"
#data_lv_t_sample = merge(metadataID,data_lv, by ='ID')
#data_lv_t_sample = data_lv_t_sample[,-1]
#data_lv_t_sample = t(data_lv_t_sample)
write.table(data_lv_t_sample, file='feature_data_with_lv_2.tsv', quote=FALSE, sep='\t', row.names = F, col.names = F)
```

## Vérification des ID 
```{r eval=TRUE, echo=TRUE}
c =0
for (i in 1:284){
  if (rownames(data_lv)[i] %in% metadataID$ID){
    c = c + 1
  }
}
c
```

## Vérification des valeurs
```{r eval=TRUE, echo=TRUE}
#dinomesomics_df = as.data.frame(dinomesomics)
#c= 0
#for (i in 1:284){
#     print(as.numeric(as.character(dinomesomics_df["ENSG00000002746.14",][i] )))
#    print(as.numeric(as.character(data_lv$ENSG00000002746.14[i])))
#  if (as.numeric(as.character(dinomesomics_df["ENSG00000002746.14",][i] ))== as.numeric(as.character(data_lv$ENSG00000002746.14[i]))){
 
#     c =c+1
#  }
#}
#c
```

# Reproduction fig 3 

## Fig 3A

```{r eval=TRUE, echo=TRUE}
Attributes5 <- Attributes4
colnames(Attributes5)[2] <- "B.Cells"
colnames(Attributes5)[29] <- "Subtype"
colnames(Attributes5)[30] <- "Type_revised"
Attributes5["Type_and_Survival"] = Attributes5$Type
Attributes5$Type_and_Survival[Attributes5$Type =='Epithelioid' & Attributes5$Survival>30] = "Epithelioid.long.survival"
Attributes5$Type_and_Survival[Attributes5$Type =='Epithelioid' & Attributes5$Survival<10] = "Epithelioid.short.survival"
Attributes5 = Attributes5[Attributes5$Type == "Epithelioid" | Attributes5$Type == "Sarcomatoid" , ]
Attributes5 = Attributes5[-c(which(Attributes5$Type_and_Survival == "Epithelioid")),]
Attributes5 = Attributes5[-c(which(Attributes5$Type_and_Survival == "Sarcomatoid" & Attributes5$Survival >10 )),]


```




## Création de la table pour ACP  cf fig 3 A

```{r eval=TRUE, echo=TRUE}
Attributes5 <- Attributes4
colnames(Attributes5)[2] <- "B.Cells"
colnames(Attributes5)[29] <- "Subtype"
colnames(Attributes5)[30] <- "Type_revised"
Attributes5["Type_and_Survival"] = Attributes5$Type
Attributes5$Type_and_Survival[Attributes5$Type =='Epithelioid' & Attributes5$Survival>30] = "Epithelioid.long.survival"
Attributes5$Type_and_Survival[Attributes5$Type =='Epithelioid' & Attributes5$Survival<10] = "Epithelioid.short.survival"
Attributes5 = Attributes5[Attributes5$Type == "Epithelioid" | Attributes5$Type == "Sarcomatoid" , ] # Removing biphasic tumors
Attributes5 = Attributes5[-c(which(Attributes5$Type_and_Survival == "Epithelioid")),] # Removing middle survival epithelioid tumors
Attributes5 = Attributes5[-c(which(Attributes5$Type_and_Survival == "Sarcomatoid" & Attributes5$Survival >10 )),] # Removing long survival sarcomatoid tumors


groupsBuenoTCGAred.2 = groupsBuenoTCGAred
groupsBuenoTCGAred.2["Type_and_survival"] = groupsBuenoTCGAred.2$Type
groupsBuenoTCGAred.2$Type_and_survival[groupsBuenoTCGAred.2$Type=="Epithelioid" & groupsBuenoTCGAred.2$Survival *12 > 30] = "Epithelioid.long.survival"
groupsBuenoTCGAred.2$Type_and_survival[groupsBuenoTCGAred.2$Type=="Epithelioid" & groupsBuenoTCGAred.2$Survival *12 < 10] = "Epithelioid.short.survival"


#groupsBuenoTCGAred= groupsBuenoTCGAred$Survival[groupsBuenoTCGAred$Type=="Epithelioid" & groupsBuenoTCGAred$Survival *12 > 30]

Test_survival_epi_sup_meta = metadata$Survival[metadata$Type=="Epithelioid" & metadata$Survival *12 > 30]

```


```{r eval=TRUE, echo=TRUE}
Attributes5 = merge(Attributes5 , metadataID , by ='sample')
rownames(Attributes5) <- Attributes5$ID
col.num <- which(colnames(dinomesomics) %in% Attributes5$ID)
data_fig_3a_transcriptome <- dinomesomics[,c(col.num)]

data_fig_3a_transcriptome <- t(data_fig_3a_transcriptome )
acp_transcriptome_fig3a <- dudi.pca(data_fig_3a_transcriptome, center = T , scale = F) #,nf=F
PCA_Coord <- as.data.frame(acp_transcriptome_fig3a$li ) 
PCA_Coord = setDT(PCA_Coord, keep.rownames = TRUE)[]
colnames(PCA_Coord)[1]<- "ID"
Attributes5 <- merge(Attributes5 , PCA_Coord , by= "ID")
summary(acp_transcriptome_fig3a )

plot(Attributes5$Axis1 , Attributes5$Axis2 , col= as.factor(Attributes5$Type_and_Survival))

#metadata$Survival == SurvBuenoTCGAred[,1]



```

# ACP avec 5 marqueurs
** A priori cet ACP a été construite avec tous les gènes **

```{r eval=TRUE, echo=TRUE}
Attributes5 = merge(Attributes5 , metadataID , by ='sample')
rownames(Attributes5) <- Attributes5$ID
col.num <- which(colnames(dinomesomics) %in% Attributes5$ID)
data_fig_3a_transcriptome <- dinomesomics[,c(col.num)]
data_fig_3a_transcriptome <- as.data.frame(data_fig_3a_transcriptome)
VISTA =  c(data_fig_3a_transcriptome["ENSG00000107738.19",])
VISTA.f = c()
for (i in 1:109){
  VISTA.f[i]=VISTA[[i]]
}
  
CD8A = c(data_fig_3a_transcriptome["ENSG00000153563.15",])
CD8A.f = c()
for (i in 1:109){
  CD8A.f[i]=CD8A[[i]]
}
VEGFR2 = c(data_fig_3a_transcriptome["ENSG00000128052.8",])
VEGFR2.f = c()
for (i in 1:109){
  VEGFR2.f[i]=VEGFR2[[i]]
}

VEGFR3 = c(data_fig_3a_transcriptome["ENSG00000037280.15",])
VEGFR3.f = c()
for (i in 1:109){
  VEGFR3.f[i]=VEGFR3[[i]]
}

PDL1 = c(data_fig_3a_transcriptome["ENSG00000120217.13",])
PDL1.f = c()
for (i in 1:109){
  PDL1.f[i]=PDL1[[i]]
}
data_acp_trans_fig3 <- data.frame(VISTA.f , CD8A.f,VEGFR2.f , VEGFR3.f , PDL1.f)
row.names(data_acp_trans_fig3) = Attributes5$ID
acp_transcriptome_fig3a <- dudi.pca(data_acp_trans_fig3, center = T , scale = F) #,nf=F
PCA_Coord <- as.data.frame(acp_transcriptome_fig3a$li ) 
PCA_Coord = setDT(PCA_Coord, keep.rownames = TRUE)[]
colnames(PCA_Coord)[1]<- "ID"
Attributes5 <- merge(Attributes5 , PCA_Coord , by= "ID")


plot(Attributes5$Axis1 , Attributes5$Axis2 , col= as.factor(Attributes5$Type_and_Survival))

#metadata$Survival == SurvBuenoTCGAred[,1]


```


# Reproduction de l'ACP de la figure 1
```{r eval=TRUE, echo=TRUE}
df_acp_fig1 <-data_lv
for (i in 1:7145){
  df_acp_fig1[,i] <- as.numeric(as.character(df_acp_fig1[,i]) )
}
str(df_acp_fig1)
acp_fig1 <- dudi.pca( df_acp_fig1 ,center = T , scale = F) #
s.label(acp_fig1$li, xax = 1, yax = 2)
acp_fig1_li_df =  as.data.frame(acp_fig1$li)
acp_fig1_li_df  = setDT(acp_fig1_li_df , keep.rownames = TRUE)[]
colnames(acp_fig1_li_df )[1]<-'ID'
acp_fig1_li_df = merge(acp_fig1_li_df , metadataID , by ='ID')
acp_fig1_li_df = acp_fig1_li_df[,-1]

acp_fig1_li_df = merge(Attributes4 , acp_fig1_li_df , by="sample")


gcol = c("blue", "red", "orange", "darkgreen")
Type_acp_fig1  = as.factor(Attributes4$Type)
#s.class(acp_fig1$li,  Type_acp_fig1  , col = gcol, xax=1, yax=2)
colnames(suptable1)[1]<-"Sample"
suptable1.2 <- merge(suptable1, metadata, by='Sample')
plot(acp_fig1_li_df$Axis1 , acp_fig1_li_df$Axis2 , col=as.factor(acp_fig1_li_df$Type) )  # ;)   Victoire
plot(suptable1.2$Dimension.1 , suptable1.2$Dimension.2 , col=suptable1.2$Type ) #OK
```



# Importation de la cohorte de replication 


```{r eval=TRUE, echo=TRUE}
Replication_data <-  read.csv("IHC_discovery_+_replication_cohorts/dataIHC.csv", sep = ",", dec="." , header = TRUE, na.strings=c("","NA"))

Vista_r = Replication_data$VISTA.membranous
CD8_r = Replication_data$CD8
Vegfr2_r = Replication_data$VEGFR2
Vegfr2_r.f = c()
for (i in 1:77){
 if (is.na(Vegfr2_r[i])==F){
  if (Vegfr2_r[i] =='0'){
    Vegfr2_r.f[i] = 0
  }
  else if (Vegfr2_r[i]=='1M'){
    Vegfr2_r.f[i] = 0.25
  }
  else if (Vegfr2_r[i]=='2M'){
    Vegfr2_r.f[i] = 0.5
  }
  else if (Vegfr2_r[i]=='3M'){
    Vegfr2_r.f[i] = 0.75
  }
  else if (Vegfr2_r[i]=='4M'){
    Vegfr2_r.f[i] = 1
  }
 }
  else{
    Vegfr2_r.f[i] = NA
  }
}


Vegfr2_r_mb = Replication_data$VEGFR2.membranous

PDL1_r = Replication_data$PDL1.Tumor
Vegfr3_r = Replication_data$VEGFR3.membranous
Vegfr3_r.f = c()
for (i in 1:77){
 if (is.na(Vegfr3_r[i])==F){
  if (Vegfr3_r[i] ==0){
    Vegfr3_r.f[i] = 0
  }
  else if (Vegfr3_r[i]==1){
    Vegfr3_r.f[i] = 0.25
  }
  else if (Vegfr3_r[i]==2){
    Vegfr3_r.f[i] = 0.5
  }
  else if (Vegfr3_r[i]==3){
    Vegfr3_r.f[i] = 0.75
  }
  else if (Vegfr3_r[i]==4){
    Vegfr3_r.f[i] = 1
  }
 }
  else{
    Vegfr3_r.f[i] = NA
  }
}

Vegfr3_r_cyto =Replication_data$VEGFR3.cytoplasmic
Vegfr3_r_cyto.f =c()
for (i in 1:77){
 if (is.na(Vegfr3_r_cyto[i])==F){
  if (Vegfr3_r_cyto[i] ==0){
    Vegfr3_r_cyto.f[i] = 0
  }
  else if (Vegfr3_r_cyto[i]==1){
    Vegfr3_r_cyto.f[i] = 0.25
  }
  else if (Vegfr3_r_cyto[i]==2){
    Vegfr3_r_cyto.f[i] = 0.5
  }
  else if (Vegfr3_r_cyto[i]==3){
    Vegfr3_r_cyto.f[i] = 0.75
  }
  else if (Vegfr3_r_cyto[i]==4){
    Vegfr3_r_cyto.f[i] = 1
  }
 }
  else{
    Vegfr3_r_cyto.f[i] = NA
  }
}

PDL1.TILS_r = Replication_data$PDL1.TILS
PDL1.TILS_r.f = c()
for (i in 1:77){
 if (is.na(PDL1.TILS_r[i])==F){
  if (PDL1.TILS_r[i] ==0){
    PDL1.TILS_r.f[i] = 0
  }
  else if (PDL1.TILS_r[i]==1){
    PDL1.TILS_r.f[i] = 1/3
  }
  else if (PDL1.TILS_r[i]==2){
    PDL1.TILS_r.f[i] = 2/3
  }
  else if ( PDL1.TILS_r[i]==3){
    PDL1.TILS_r.f[i] = 1
  }
 
 }
  else{
    PDL1.TILS_r.f[i] = NA
  }
}

replication_acp = data.frame(Vista_r, Vegfr2_r_mb ,  Vegfr3_r.f , CD8_r,  PDL1.TILS_r.f, PDL1_r) #Vegfr3_r_cyto.f ,Vegfr2_r.f , 
rownames(replication_acp)=Replication_data$X
replication_acp = replication_acp[complete.cases(replication_acp),]

acp_rep <- dudi.pca(replication_acp , center = T  , scale = F)
acp_rep_li = as.data.frame(acp_rep$li)
acp_rep_li = setDT(acp_rep_li, keep.rownames = TRUE)[]
colnames(acp_rep_li)[1] <- "X"
Attributes_replication = merge(Replication_data, acp_rep_li , by="X")

plot(Attributes_replication$Axis1, Attributes_replication$Axis2 , col = as.factor(Attributes_replication$GRP)) # Victoire
```
